code/modules/mob/dead/observer/observer.dm

```diff
@@ -116,6 +116,10 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		loc = NewLoc
 		for(var/obj/effect/step_trigger/S in NewLoc)
 			S.Crossed(src)
+		var/area/new_area = get_area(loc)
+		if(new_area.blacklisted)
+			src << "You can't move around here!"
+			loc = pick(latejoin)
```

Comment:
what about just putting them back on their old loc?
eg: before ```if(NewLoc)``` put ```var/oloc = loc```
its less fucky for the player.